src/protocols/rdp/channels/rdpsnd/rdpsnd-messages.c

```diff
@@ -296,7 +296,7 @@ void guac_rdpsnd_wave_info_handler(guac_rdp_common_svc* svc,
     rdpsnd->next_pdu_is_wave = TRUE;
 
     /* Reset audio stream if format has changed */
-    if (audio != NULL)
+    if (audio != NULL && format < sizeof(rdpsnd->formats))
```

Comment:
Looks good - very straightforward. The only nit I'd like to pick is logging: rather than silently trudge forward, I think we should log that something isn't right, just as in #274. If we encounter a faulty RDP server and this occurs, it would be good to know that it occurred, both for debugging and for reporting upstream.

Lacking that, this sort of failure would no doubt result in JIRA issues / mailing list threads regarding mysterious lack of sound, and it would take some involved digging before things could be established to not be a guac issue.

Reply:
Okay, took a stab at a log message - probably needs some tweaking, so let me know what needs to be changed.